domain/consensus/utils/utxo/utxo_entry.go

```diff
@@ -0,0 +1,38 @@
+package utxo
+
+import "github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+
+type utxoEntry struct {
+	amount          uint64
+	scriptPublicKey []byte
+	blockBlueScore  uint64
+	isCoinbase      bool
+}
+
+// NewUTXOEntry creates a new utxoEntry representing the given txOut
+func NewUTXOEntry(amount uint64, scriptPubKey []byte, isCoinbase bool, blockBlueScore uint64) externalapi.UTXOEntry {
+	scriptPubKeyClone := make([]byte, len(scriptPubKey))
+	copy(scriptPubKeyClone, scriptPubKey)
+	return &utxoEntry{
+		amount:          amount,
+		scriptPublicKey: scriptPubKeyClone,
+		blockBlueScore:  blockBlueScore,
+		isCoinbase:      isCoinbase,
+	}
+}
+
+func (u utxoEntry) Amount() uint64 {
```

Comment:
Use pointer receiver, otherwise you'd copy the entire entry for each method call